lib/instrumentation/undici.js

```diff
@@ -133,17 +144,25 @@ module.exports = function addUndiciChannels(agent, undici, modName, shim) {
     const url = new URL(urlString)
 
     const name = NAMES.EXTERNAL.PREFIX + url.host + url.pathname
-    const segment = shim.createSegment(name, recordExternal(url.host, 'undici'), parentSegment)
+    let segment
+    if (request[SYMBOLS.SEGMENT]) {
+      segment = request[SYMBOLS.SEGMENT]
+      segment.name = name
```

Comment:
nice, glad this just works by reassigning name after it was created as unknown above

Reply:
Yeah, though I am a little worried that an async harvest cycle might report this segment prior to this update -- what do you think?

Reply:
I'm not sure I understand what you mean.  we only remove segments when the response headers have been sent.

Reply:
Oh, well that's convenient. As for not understanding, it's just that I don't completely understand how harvest cycles work, so I was worried the segment would be removed before it was properly decorated.

Reply:
Segments are tied to the transaction lifecycle, so they won't generate their data or be cleaned up (at least from the transaction) until the transaction has ended. Although, these symbol usages are a bit different in that they propagate separate from that. If they happened to propagate after the transaction was ended then the segment updates would indeed be lost but then we also have other problems (which we actually might have, may or may not be related to this style of propagation tho).